jerry-ext/args/arg-transform-functions.c

```diff
@@ -0,0 +1,444 @@
+
+  if (size == 0)
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "The size of the buffer is not large enough.");
+  }
+
+  return jerry_create_undefined ();
+}
+
+jerry_value_t
+jerryx_arg_transform_string_optional (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (jerry_value_is_undefined (js_arg))
+  {
+    return jerry_create_undefined ();
+  }
+
+  jerry_value_t to_string = jerry_value_to_string (js_arg);
+
+  if (jerry_value_has_error_flag (to_string))
+  {
+    jerry_release_value (to_string);
+
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It can not be converted to a string.");
+  }
+
+
+  jerry_length_t dest_len = (jerry_length_t) iterator_p->c_arg_p->extra_info;
+  char *dest = jerryx_arg_iterator_get_dest (iterator_p);
+
+  jerry_size_t size = jerry_string_to_char_buffer (to_string,
+                                                   (jerry_char_t *) dest,
+                                                   dest_len);
+
+  if (size == 0)
+  {
+    jerry_release_value (to_string);
+
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "The size of the buffer is not large enough.");
+  }
+
+  jerry_release_value (to_string);
+
+  return jerry_create_undefined ();
+}
+
+jerry_value_t
+jerryx_arg_transform_string (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  jerry_value_t to_string = jerry_value_to_string (js_arg);
+
+  if (jerry_value_has_error_flag (to_string))
+  {
+    jerry_release_value (to_string);
+
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It can not be converted to a string.");
+  }
+
+
+  jerry_length_t dest_len = (jerry_length_t) iterator_p->c_arg_p->extra_info;
+  char *dest = jerryx_arg_iterator_get_dest (iterator_p);
+
+  jerry_size_t size = jerry_string_to_char_buffer (to_string,
+                                                   (jerry_char_t *) dest,
+                                                   dest_len);
+
+  if (size == 0)
+  {
+    jerry_release_value (to_string);
+
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "The size of the buffer is not large enough.");
+  }
+
+  jerry_release_value (to_string);
+
+  return jerry_create_undefined ();
+}
+
+/* function transform functions. */
+
+jerry_value_t
+jerryx_arg_transform_function (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (!jerry_value_is_function (js_arg))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It is not a function.");
+  }
+
+  jerry_value_t *func_p = jerryx_arg_iterator_get_dest (iterator_p);
+  *func_p = jerry_acquire_value (js_arg);
```

Comment:
Hmm, I wonder if we should also `jerry_release_value(*func_p)` first.
I suspect this will probably be used in some callback-setter function.

If we don't release here, people will have to worry about either adding a custom transform (by wrapping this one) or manually take care of the releasing.
I think because this is such a basic use case, the util should take care of this.

However, if we release it, it means that the destination `jerry_value_t` cannot be junk / it has to be initialized properly or bad things will happen.
Personally, I think that's probably an acceptable consequence and probably no problem in most cases.

Alternatively, we could add another arg to `jerryx_arg_function` (for example`JERRYX_AUTO_RELEASE` vs `JERRYX_NO_AUTO_RELEASE`). I suspect most use cases will require releasing of what is already there.

@janjongboom thoughts? I think you had a use case for this, which did involve releasing the old `jerry_value_t`.